### PR TITLE
Docs on the new ARM template for deploying on Azure

### PR DESCRIPTION
**Pull request by ttmc**
We now have two ARM templates on the Azure Quickstart Templates repo so I added a new page to the docs to help users with the new one. (The old one is still there, for now.)

In the process of testing it, I discovered some issues with the new ARM template (and accompanying bash script). I fixed those issues and pushed the fixes to the Azure Quickstart Templates repo: see https://github.com/Azure/azure-quickstart-templates/pull/2785 

That PR (2785) should be merged before we merge this PR.

### CODE REVIEW

File: docs/server/source/cloud-deployment-starter-templates/new-azure-template.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+One can deploy a BigchainDB node on Azure using the template in the `bigchaindb-on-ubuntu` directory of Microsoft's `azure-quickstart-templates` repository on GitHub:
+
+1. Go to [the `bigchaindb-on-ubuntu` directory in that repository](https://github.com/Azure/azure-quickstart-templates/tree/master/bigchaindb-on-ubuntu).
+
+2. Click the button labelled **Deploy to Azure**.
+
+3. If you're not already logged in to Microsoft Azure, then you'll be prompted to login. If you don't have an account, then you'll have to create one.
+
+4. One you are logged in to the Microsoft Azure Portal, you should be taken to a form titled **BigchainDB**. There are some notes to help with filling in that form at the bottom of this page.
```
Comment (sohkai) [inline docs/server/source/cloud-deployment-starter-templates/new-azure-template.md]: 4. On**c**e ...

Comment (sohkai) [inline docs/server/source/cloud-deployment-starter-templates/new-azure-template.md]: Might be nice to add a link to the section below, e.g. Some notes to help with filling in this form are available [below](#notes-on-the-blockchain-template-form-fields).

File: docs/server/source/cloud-deployment-starter-templates/new-azure-template.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+5. Deployment takes a few minutes. You can follow the notifications by clicking the bell icon at the top of the screen. When done, you should see a notification saying "Deployment to resource group '[your resource group name]' was successful." The install script (`init.sh`) installed RethinkDB, configured it using the default RethinkDB config file, and ran it. It also used `pip` to install [the latest `bigchaindb` from PyPI](https://pypi.python.org/pypi/BigchainDB).
+
+6. Find out the public IP address of the virtual machine in the Azure Portal. Example: `40.69.87.250`
```
Comment (sohkai) [inline docs/server/source/cloud-deployment-starter-templates/new-azure-template.md]: Maybe this is obvious from the Azure interface, but just wanted to check that most people will find this step obvious.

File: docs/server/source/cloud-deployment-starter-templates/new-azure-template.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+7. ssh in to the virtual machine at that IP address, e.g. `ssh adminusername@40.69.87.250` (where `adminusername` should be replaced by the Admin Username you entered into the form, and `40.69.87.250` should be replaced by the IP address you found in the last step).
```
Comment (sohkai) [inline docs/server/source/cloud-deployment-starter-templates/new-azure-template.md]: Might be nicer as `ssh <admin-username>@<machine-ip>` and then give examples for the two (e.g. `40.69.87.250` as an example of `<machine-ip>`).

File: docs/server/source/cloud-deployment-starter-templates/new-azure-template.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+**Resource group** - You can use an existing resource group (if you have one) or create a new one named whatever you like, but avoid using fancy characters in the name because Azure might have problems if you do.
+
+**Location** is the Microsoft Azure data center where you want the BigchainDB node to run. Pick one close to where you are located.
+
+### SETTINGS
+
+You can use whatever **Admin\_username** and **Admin\_password** you like (provided you don't get too fancy). It will complain if your password is too simple. You'll need these later to `ssh` into the virtual machine.
+
+**Dns\_label\_prefix** - Once your virtual machine is deployed, it will have a public IP address and a DNS name (hostname) something like `DNSprefix.northeurope.cloudapp.azure.com`. The `DNSprefix` will be whatever you enter into this field.
```
Comment (sohkai) [inline docs/server/source/cloud-deployment-starter-templates/new-azure-template.md]: Not 100% sure what we're using as conventions for substitutable strings in the docs, but it's nice to have some sort of delimiter to visually clarify them, e.g. `<DNSprefix>.northeurope.cloudapp.azure.com`.

File: docs/server/source/cloud-deployment-starter-templates/new-azure-template.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+**Virtual\_machine\_size** - This should be one of Azure's standard virtual machine sizes, such as `Standard_D1_v2`. There's a [list of virtual machine sizes in the Azure docs](https://docs.microsoft.com/en-us/azure/virtual-machines/virtual-machines-windows-sizes?toc=%2fazure%2fvirtual-machines%2fwindows%2ftoc.json).
+
```
Comment (sohkai) [inline docs/server/source/cloud-deployment-starter-templates/new-azure-template.md]: Extra newline?

File: docs/server/source/cloud-deployment-starter-templates/new-azure-template.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+8. You should be prompted for a password. Enter the Admin Password you entered into the form.
+
+9. Configure BigchainDB using the default BigchainDB settings: `bigchaindb -y configure`
+
+10. Run BigchainDB: `bigchaindb start`
+
+BigchainDB should now be running on the Azure VM.
+
+Remember to shut everything down when you're done (via the Azure Portal), because it generally costs money to run stuff on Azure.
```
Comment (sohkai) [inline docs/server/source/cloud-deployment-starter-templates/new-azure-template.md]: :trollface:

File: docs/server/source/cloud-deployment-starter-templates/new-azure-template.md (hunk 2 above)
Comment (ttmc) [inline docs/server/source/cloud-deployment-starter-templates/new-azure-template.md]: Sadly, the virtual machine's IP address is not at all obvious. It takes some digging to find it. (There are many services launched and the virtual machine is only one of them. In fact, another one of the services is the static IP address and one could look there instead...)

**Pull request merged**